### ISSUE TITLE
Hugo Server/LiveReload breaks after modifying Resized image resource.

### ISSUE THREAD

**Issue by Ziflin** [gohugoio/hugo#4988]
_Hugo Version:_ v0.45/extended windows/amd64 BuildDate: unknown
_OS:_ Windows 10

I have a simple test setup of:
```
content/
  posts/
      post1/
        index.md
        image.jpg
```
I'm using [Krita](https://krita.org/en/) (open source image editor) to modify `image.jpg` while `hugo server` is running and I have a browser open to  `.../post1/`. When I re-save a change to the image in Krita, I get two notifications from Hugo:

```
Change detected, rebuilding site
2018-07-23 09:33:20.107 -0400
Source changed "D:\\Websites\\Hugo\\test\\content\\posts\\post1\\image.jpg": CREATE
Source changed "D:\\Websites\\Hugo\\test\\content\\posts\\post1\\image.jpg.Hp9760": RENAME
Total in 56 ms
```
I assume Krita is saving to a temporary file (`image.jpg.Hp9760`) and the renaming it to `image.jpg`. I would have expected that `CREATE` to be a `DELETE` though.

I also have a simple shortcode for `image.html` setup to generate a resized version of the original image which is used in `post1/index.md`:
```
{{ $original := $.Page.Resources.GetMatch ( .Get 0 ) }}
{{ with $original.Resize "600x" }}
<div>
    <img src="{{ .RelPermalink }}">
</div>
{{ end }}
```
If I change the `<img>` to use `$original.RelPermalink`, everything works as I expect it to. I can modify `image.jpg` and resave it over and over and the live reload continues to refresh properly.

If I instead have it as shown above using the resized image, the first image change and re-save works, but after that it breaks and live-reload stops working. There are no errors in the log until I `touch` the shortcode file and then I get:

```
Change detected, rebuilding site
2018-07-23 09:46:08.109 -0400
Template changed "D:\\Websites\\Hugo\\test\\themes\\hexed\\layouts\\shortcodes\\image.html": WRITE
ERROR 2018/07/23 09:46:08 error processing shortcode "shortcodes/image.html" for page "posts\\post1\\index.md": template: shortcodes/image.html:2:17: executing "shortcodes/image.html" at <$original.Resize>: can't evaluate field Resize in type resource.Resource
Total in 19 ms
ERROR 2018/07/23 09:46:08 Failed to rebuild site: logged 1 error(s)
```

So basically it seems to have lost the `$original` resource that was _previously_ working fine. If I stop and restart the server, everything works as expected. So it's something to do with the server file monitoring and resources getting out of sync with what's actually there when using the image processing calls.

**Comment by nonumeros** [comment]
`can't evaluate field Resize in type resource.Resource`
did you check
https://discourse.gohugo.io/t/cant-evaluate-field-resize-in-type-resource-resource/9953

**Comment by bep** [comment]
You should try this with the latest Hugo.

**Comment by Ziflin** [comment]
@bep Thanks! I'll try to give it a shot tomorrow.

**Comment by Ziflin** [comment]
@bep I just tested it with 0.47.1 and it's still having the same problem. It will auto-refresh properly the first time I resave the image, but if I make a second change to it, then it stops updating and I have to restart the server to see it. 
I'm guessing it still has something to do with the way Krita and Photoshop save to a new file and then move/rename the file back to the original as it does seem to work with MSPaint which just overwrites the original (Hugo logs a "WRITE" change vs "CREATE" and "RENAME" with Krita).
